plugin/plaintextmodel.cpp

```diff
@@ -117,6 +121,11 @@ void PlainTextModel::setSource(const QUrl &source)
                 m_status = Error;
             } else {
                 m_textStream.setDevice(&m_file);
+                QByteArray ba = m_file.readAll();
```

Comment:
I think it is better to detect encoding of the whole file content.

Reply:
Why? Some character around 5MB position changing the result?

Reply:
I think you are right

Reply:
You could mmap the file rather than streaming the data, that should ensure there's enough data available without potentially allocating megabytes on the heap to scan the first handful of characters.

Reply:
> You could mmap the file rather than streaming the data, that should ensure there's enough data available without potentially allocating megabytes on the heap to scan the first handful of characters.

I need the streaming data to establish the right codec

Reply:
Mapping the data, instead of reading them, will allow to use the data (in the detection library for instance) without allocating the memory: https://doc.qt.io/qt-5/qfiledevice.html#map From my understanding, in that case, the actual file for storage is used when accessing the data and not a copy of the data into memory.

Reply:
Done